### PR TITLE
Add a proper `command_timeout` support to Remoter

### PR DESCRIPTION
**Pull request by fruch**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (`hydra unit-tests`)
- ~~[ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by fruch** [comment]
it passed a run with 3h

trying now with a run with 4h:
http://jenkins.scylladb.com/view/staging/job/scylla-staging/job/empty/job/longevity/job/longevity-100gb-4h-test/1

File: docker/env/version [hunk 1]
```diff
@@ -1 +1 @@
-0.38
+0.39
```
Comment (bentsi) [inline docker/env/version]: I think when testing new versions we should use some testing version, like 0.39a, 0.39b,...0.39z

Comment (fruch) [reply docker/env/version]: next time I'll try to remember that...

**Comment by fruch** [comment]
@bentsi @roydahan the 4h finished with this after almost 5h and with nice shade of blue....

**Pull request merged**